lib/bolt/transport/base.rb

```diff
@@ -68,6 +69,33 @@ def with_events(target, callback)
         result
       end
 
+      def provided_features
+        []
+      end
+
+      def default_input_method
+        'shell'
+      end
+
+      def select_implementation(target, task)
+        impl =task.select_implementation(target.run_on_target, provided_features)
+        impl['input_method'] ||= default_input_method
+        impl
+      end
+
+      def add_target_param(target, task, params)
+        # TODO: How should metadata impact this?
```

Comment:
Metadata seems to be more "controllable" (and visible) to the user, so I think "prioritizing" that makes the most sense:
```
if !target.remote? && task.remote?
    raise Bolt::Node::RemoteError.new('Remote tasks can only be run on remote targets')
end
```
Raising an error if the target is remotable and the task doesn't specify it could be confusing if users don't know where the 'remotable' target is coming from.

Reply:
on the other hand running a non-remote task against a remote target seems incredibly dangerous.

Reply:
I'm not sure I understand what a task be "remotable" is enough to comment. For right now it just seems like "the user said in the metadata it can be run on remote targets", and if they *don't* specify that they still might be able to run on remote targets. 

I think we can raise a warning in that case, but limiting the user to only tasks that are remote seems like it makes using remote tasks much harder.

Reply:
I think the general statement is that a remoteable task is one that does something meaningful with the `_target` metaparam. 

IE the service task is not remoteable. If I ran it targeting an aws account it would stop a service on the run-on node rather then do anything to the real device.